sdk/typescript/src/types/common.ts

```diff
@@ -9,3 +9,35 @@ export type ObjectOwner =
   | { ObjectOwner: SuiAddress }
   | 'Shared'
   | 'Immutable';
+
+
+const TX_DIGEST_BASE64_LENGTH = 44;
+// taken from https://rgxdb.com/r/1NUN74O6
+const VALID_BASE64_REGEX =
+  /^(?:[a-zA-Z0-9+\/]{4})*(?:|(?:[a-zA-Z0-9+\/]{3}=)|(?:[a-zA-Z0-9+\/]{2}==)|(?:[a-zA-Z0-9+\/]{1}===))$/;
+
+export function isValidTransactionDigest(value: string): value is TransactionDigest {
+  return value.length === TX_DIGEST_BASE64_LENGTH
+    && VALID_BASE64_REGEX.test(value);
+}
+
+// TODO - can we automatically sync this with rust length definition?
+const SUI_ADDRESS_LENGTH = 16;
```

Comment:
```suggestion
// Source of truth is
// https://github.com/MystenLabs/sui/blob/acb2b97ae21f47600e05b0d28127d88d0725561d/crates/sui-types/src/base_types.rs#L67 
// which uses the Move account address length 
// https://github.com/move-language/move/blob/67ec40dc50c66c34fd73512fcc412f3b68d67235/language/move-core/types/src/account_address.rs#L23 .

const SUI_ADDRESS_LENGTH = 20;
```
The correct length is 20 instead of 16.

In general we should add a comment about the source of truth when we are using constants defined outside the SDK.